File: src/main/java/me/kuku/yuq/job/MiHoYoJob.java
package me.kuku.yuq.job;

import com.IceCreamQAQ.Yu.annotation.Cron;
import com.IceCreamQAQ.Yu.annotation.JobCenter;
import me.kuku.yuq.entity.MiHoYoEntity;
import me.kuku.yuq.logic.MiHoYoLogic;
import me.kuku.yuq.service.MiHoYoService;

import javax.inject.Inject;
import java.io.IOException;
import java.util.List;

@JobCenter
public class MiHoYoJob {

	@Inject
	private MiHoYoLogic miHoYoLogic;
	@Inject
	private MiHoYoService miHoYoService;

	@Cron("At::d::08:30")
	public void genShinSign(){
		List<MiHoYoEntity> list = miHoYoService.findAll();
		list.forEach(entity -> {
			try {
				miHoYoLogic.sign(entity);
			} catch (IOException e) {
				e.printStackTrace();
			}
		});
	}
}

File: src/main/kotlin/me/kuku/yuq/job/MiHoYoJob.kt
package me.kuku.yuq.job

import com.IceCreamQAQ.Yu.annotation.Cron
import com.IceCreamQAQ.Yu.annotation.JobCenter
import me.kuku.yuq.entity.MiHoYoService
import me.kuku.yuq.entity.Status
import me.kuku.yuq.logic.MiHoYoLogic
import org.springframework.stereotype.Component
import javax.inject.Inject

@JobCenter
@Component
class MiHoYoJob (
    private val miHoYoService: MiHoYoService
) {


    @Cron("05:13")
    suspend fun genShinSign() {
        val list = miHoYoService.findAll().filter { it.config.sign == Status.ON }
        for (miHoYoEntity in list) {
            MiHoYoLogic.sign(miHoYoEntity)
        }
    }

}